### ISSUE TITLE
Popups and child-windows are drawn behind parent windows in certain situations

### ISSUE THREAD

**Issue by tcld** [qtile/qtile#1893]
# Issue description
In some situations I noticed that popups or child-windows are drawn (or moved) behind their respective parents; for child-windows this can be quite a problem in case the child is blocking, since then the software might be completely unresponsive.

Places where I have noticed this and that I remember:
1. Virtualbox (fullscreen) (the main window, not a client): Context menus on VM-list open in the background.
2. Veracrypt (floating) "mount" child-window: The window either spawns in the back or is moved there once it loses focus.

There have been some other cases, but due to testing PR #1870, which I initially expected to be the cause for this, I haven't taken note of all of them.
Some software definitely has working popup-windows, so Virtualbox might be doing something weird.

# Qtile version

https://github.com/qtile/qtile/commit/c6b80e9444e9b4e5dc2f52ca1954dc502c45f2eb
(newer versions not tested yet)

# Configuration

I'll only include one snippet here, since verything else (setting up Keys and Widgets) most likely has nothing to do with the problem.

```python
@hook.subscribe.client_new
def modify_window(client):
    if (client.window.get_wm_transient_for() or client.window.get_wm_type() in floating_types):
        client.floating = True
```

On top of this I am currently using fake screens instead of normal screens.

It would be nice if somebody could confirm this, since I am not entirely sure this isn't all caused by my own PR.

Related to #1870, which could fix at least the described issue with child-windows.

**Comment by ramnes** [comment]
Yes, I've noticed this with the Gimp export floating windows also: when you export as PNG, the second floating window (PNG settings) appears behind the first one.

**Comment by cr1ogen** [comment]
In Steam too, in floating layout

**Comment by ramnes** [comment]
Not sure if it's related or not, but if you click a button on that Gimp floating window for PNG settings and there's the other floating window behind, then the button won't get clicked and the other window will get focus. Really weird.

**Comment by cr1ogen** [comment]
Yes, that also happens to me on Steam, in the chat pop-up window

**Comment by cr1ogen** [comment]
But, for example in OBS Studio floating windows do not present problems

**Comment by tcld** [comment]
I also have a few programs that seem to work fine. Steam is a bit weird, though - especially the menus don't open reliably for me.

If you aren't running the git master, please include the version you are running. I don't remember any of this behavior in 0.15.1, so it might have been introduced since then.

**Comment by cr1ogen** [comment]
qtile --version
0.16.2.dev26+g10645392

**Comment by luis-barrera** [comment]
Also the same problem with GIMP

**Comment by tcld** [comment]
Might be related to PR #1710 (commit https://github.com/qtile/qtile/commit/0f886606e1f35226b325f5405d43946976eae831 ).

**Comment by tcld** [comment]
I found a minimal change in the code that seems to fix this problem:
[Here](https://github.com/qtile/qtile/blob/master/libqtile/layout/floating.py#L237) the following call can be added:

`client.window.configure(**dict(x=client.x, y=client.y, width=client.width, height=client.height))`

I have not created a PR, since I don't understand why this fixes the issues I observed. Also, if this truly is the fix we need, we might have to take care of the other special cases handled by the floating layout, too. Maybe somebody can shed some light on this?

@m-col
